### PR TITLE
PDCT-697 warn_user_exit_family_form_no_save

### PR DESCRIPTION
**Pull request by asacristani**

### CODE REVIEW

**Comment by asacristani** [comment]
Modal screenshot:
image

**Comment by asacristani** [comment]
Browser modal for external navigation (document request):
image

File: src/tests/components/forms/FamilyForm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  new Promise((resolve) => process.nextTick(resolve))
+
+// useBlocker only can be used in a router context
+jest.mock('react-router-dom', (): unknown => ({
+  ...jest.requireActual('react-router-dom'),
+  useBlocker: jest.fn(),
+}))
+
+// API calls
+jest.mock('axios', () => {
```
Comment (diversemix) [inline src/tests/components/forms/FamilyForm.test.tsx:24]: I'd be interested to know if we still need this mock?

**Review by diversemix — APPROVED** [review]
Looking good - did you want to merge this now and raise 2 tickets:
- Add testing the model when not saved
- Add complete testing for the FamilyForm

File: src/tests/components/forms/FamilyForm.test.tsx (hunk 1 above)
Comment (asacristani) [reply src/tests/components/forms/FamilyForm.test.tsx]: You've been completely right

**Pull request merged**